pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1022,6 +1025,28 @@ public ZooKeeper getZkClient() {
         return this.localZooKeeperConnectionProvider.getLocalZooKeeper();
     }
 
+    // get bookkeeper's zookeeper
+    public ZooKeeper getBookieZkClient() {
+        ZooKeeper zkClient = null;
+        String bookkeeperMetadataServiceUri = config.getBookkeeperMetadataServiceUri();
+        if (StringUtils.isBlank(bookkeeperMetadataServiceUri)) {
+            bookkeeperMetadataServiceUri = PulsarService.bookieMetadataServiceUri(config);
+        }
+
+        URI uri = URI.create(bookkeeperMetadataServiceUri);
+        String path = ZkUtils.trimLedgersDefaultRootPath(uri.getPath());
+        String bookieZkConnect = StringUtils.replace(uri.getAuthority(), ";", ",") + path;
+
+        int zkTimeout = (int) config.getZooKeeperSessionTimeoutMillis();
+        try {
+            zkClient = ZooKeeperClient.newBuilder().connectString(bookieZkConnect)
+                    .sessionTimeoutMs(zkTimeout).build();
+        } catch (Exception e) {
+            LOG.error("Error creating bookie zookeeper client with {} for bookie.", bookieZkConnect, e);
```

Comment:
in this case we are returning `null` 
should we throw an exception and fail ?

Reply:
yeah, done.